dace/codegen/targets/rtl.py

```diff
@@ -0,0 +1,544 @@
+        if(DEBUG){{
+            std::cout << "feed new element" << std::endl;
+        }}
+        {inputs}
+        model->valid_i = 1;
+    }}
+
+    // export element
+    if(model->valid_o == 1){{
+        if(DEBUG){{
+            std::cout << "export element" << std::endl;
+        }}
+        {outputs}
+        model->ready_i = 1;
+    }}
+    // check if valid_o and ready_i have been asserted at the rising clock edge -> output write handshake
+    if (model->ready_i == 1 && model->valid_o == 1){{
+        write_output_hs = true;
+    }}
+
+    // positive clock edge
+    model->clk_i = !model->clk_i;
+    model->eval();
+
+    // report internal state
+    if(DEBUG){{
+        VL_PRINTF("[t=%lu] clk_i=%u rst_i=%u valid_i=%u ready_i=%u valid_o=%u ready_o=%u \\n", main_time, model->clk_i, model->rst_i, model->valid_i, model->ready_i, model->valid_o, model->ready_o);
+        VL_PRINTF("{internal_state_str}\\n", {internal_state_var});
+        std::cout << std::flush;
+    }}
+
+    // check if valid_i and ready_o have been asserted at the rising clock edge
+    if (read_input_hs){{
+        // remove valid_i flag
+        model->valid_i = 0;
+        if(DEBUG){{
+            std::cout << "remove read_input_hs flag" << std::endl;
+        }}
+        read_input_hs = false;
+    }}
+
+    // check if valid_o and ready_i have been asserted at the rising clock edge
+    if (write_output_hs){{
+        // remove ready_i flag
+        model->ready_i = 0;
+        if(DEBUG){{
+            std::cout << "remove write_output_hs flag" << std::endl;
+        }}
+        write_output_hs = false;
+    }}
+
+    // negative clock edge
+    model->clk_i = !model->clk_i;
+    model->eval();
+}}
+
+// report internal state
+if(DEBUG){{
+    VL_PRINTF("[t=%lu] clk_i=%u rst_i=%u valid_i=%u ready_i=%u valid_o=%u ready_o=%u \\n", main_time, model->clk_i, model->rst_i, model->valid_i, model->ready_i, model->valid_o, model->ready_o);
+    VL_PRINTF("{internal_state_str}\\n", {internal_state_var});
+    std::cout << std::flush;
+}}
+
+// final model cleanup
+model->final();
+
+// clean up resources
+delete model;
+model = NULL;
+
+if(DEBUG){{
+    std::cout << "SIM {name} END" << std::endl;
+}}
+"""
+
+    RTL_HEADER = """\
+module {name}
+{parameters}
+( input                  clk_i  // convention: clk_i clocks the design
+, input                  rst_i  // convention: rst_i resets the design
+, input                  valid_i
+, input                  ready_i
+{inputs}
+, output reg             ready_o
+, output reg             valid_o
+{outputs}
+);
+"""
+
+    RTL_FOOTER = """\
+endmodule
+"""
+
+
+def check_issymbolic(iterator: iter, sdfg):
+    for item in iterator:
+        # catch symbolic (compile time variables)
+        if symbolic.issymbolic(item, sdfg.constants):
+            raise RuntimeError(
+                "Please use sdfg.specialize to specialize the symbol in expression: {}"
+                .format(item))
```

Comment:
I think this should be a `ValueError`, and make the error message a bit more descriptive, like "value of {} must be constant to generate hardware, please use sdfg.specialize..."